scripts/who-am-i/src/contextual-build-config.ts

```diff
@@ -0,0 +1,73 @@
+export interface Config {
+    /**
+     * Files to be excluded from transforming.
+     */
+    transformExcludes?: RegExp[];
+
+    /**
+     * Module overrides.
+     */
+    moduleOverrides?: Record<string, string>;
+}
+
+export interface ConfigContext {
+    /**
+     * Is performing minified-building.
+     */
+    minified?: boolean;
+
+    /**
+     * The build mode id.
+     */
+    mode?: string;
+
+    /**
+     * The platform id.
+     */
+    platform?: string;
+
+    /**
+     * Is build target support WebAssembly.
+     */
+    wasm?: boolean | 'fallback';
+
+    /**
+     * Entries.
+     */
+    entries?: string[];
+}
+
+export type ConfigFunction = (context: ConfigContext) => Promise<Config> | Config;
+
+export default (context: ConfigContext): Config => {
+    const config = {
+        transformExcludes: getTransformExcludes(context),
+        moduleOverrides: getModuleOverrides(context),
+    };
+
+    return config;
+};
+
+function getTransformExcludes (_context: ConfigContext): NonNullable<Config['transformExcludes']> {
+    return [
+        /node_modules[/\\]@cocos[/\\]ammo/,
+        /node_modules[/\\]@cocos[/\\]cannon/,
+    ];
```

Comment:
Here we can prevent some external dependencies from being transformed. @JayceLai  Notable changes.